packages/components/MenuButton/src/MenuButton.tsx

```diff
@@ -82,9 +81,23 @@ export const MenuButton = compose<MenuButtonType>({
         <Slots.button />
         {context.showContextualMenu && (
           <Slots.contextualMenu>
-            {menuItems.map((menuItem) => (
-              <ContextualMenuItem key={menuItem.itemKey} {...menuItem} />
-            ))}
+            {menuItems.map((menuItem) => {
+              const { hasSubmenu, submenu, showSubmenu, componentRef, submenuItems, ...items } = menuItem;
```

Comment:
What happens if multiple submenus say that they're showing?

Reply:
If there's multiple submenues it should be separate onShow, onHover, etc. methods for each Submenu. I checked it, but didn't include it on the Test page. Maybe it would be good to add :)

Reply:
The question was more about what happens if multiple Submenus have their showSubmenu set to true. If you'll be handling it internally that's fine, but it's something to think about if the client is going to be setting the visibility of the submenus.

Reply:
I implemented both version, with default methods and for those who wants to use custom ones